lib/pbench/test/unit/server/query_apis/test_datasets_update.py

```diff
@@ -240,14 +240,17 @@ def test_query_owner_publish(
         is_admin = build_auth_header["header_param"] == HeaderTypes.VALID_ADMIN
         if not HeaderTypes.is_valid(build_auth_header["header_param"]):
             expected_status = HTTPStatus.UNAUTHORIZED
-        elif not bool(query_json):
+        elif not query_json:
             expected_status = HTTPStatus.BAD_REQUEST
-        elif not is_admin and owner:
+            ds_name = "drb"
```

Comment:
This method of managing the `ds_name` value seems kind of arcane and fragile.  How about setting `ds_name` as a parameter:
```
    @pytest.mark.parametrize("ds_name", ("drb", "test"))
    @pytest.mark.parametrize("owner", ("drb", "test", None))
    @pytest.mark.parametrize("access", ("public", "private", None))
    def test_query_owner_publish(
        self,
        access,
        attach_dataset,
        build_auth_header,
        client,
        create_drb_user,
        create_user,
        ds_name,
        get_document_map,
        monkeypatch,
        owner,
        server_config,
    ):
```
I _think_ that `pytest.mark.parametrize()` is composable like that...give it a try.  If not, perhaps this would work better than listing out all the permutations individually:
```
    @pytest.mark.parametrize(
        "ds_name,owner,access",
        [
            (ds, o, a)
            for ds in ("drb", "test")
            for o in ("drb", "test", None)
            for a in ("public", "private", None)
        ],
    )
    def test_query_owner_publish(
        self,
        access,
        attach_dataset,
        build_auth_header,
        client,
        create_drb_user,
        create_user,
        ds_name,
        get_document_map,
        monkeypatch,
        owner,
        server_config,
    ):
```